common/azure-common/src/main/java/com/azure/common/implementation/RestProxy.java

```diff
@@ -401,6 +404,12 @@ private static Exception instantiateUnexpectedException(Class<? extends ServiceR
         Class<? extends Response<?>> cls = (Class<? extends Response<?>>) TypeUtil.getRawClass(entityType);
         if (cls.equals(Response.class)) {
             cls = (Class<? extends Response<?>>) (Object) ResponseBase.class;
+        } else if (cls.equals(PagedResponse.class)) {
```

Comment:
Here we're using `PagedResponseBase` concrete class if user mark the proxy method to return `PagedResponse` interface, which is good and we've tests for it.

Can we have a test where user returns his own type that extends from `PagedResponseBase` with `@WireReturnType` as type extending from `Page` ? Basically a test demonstrate - 

1. lib dev can define his own page type (extending from our `Page interface`) with additional properties, not just `items` and `nextlink`.

2. lib dev can have his own paged response type with some additional getters (which can be populated from additional properties in his above Page).



I thought this will be helpful at later point to understand/remember what our runtime can offer in terms of pagination. Do you think it make sense?

Reply:
I understand that custom page type in 1 and custom paged response type in 2 cannot be public, so in convince layer lib-dev still have to do a `map` to translate custom response type to customer facing response type, I guess we all implcitlty agree to it :) .